### DISCUSSION TITLE
prisma.$queryRaw is not a function

### DISCUSSION THREAD

**Discussion by luminous8** [prisma/prisma#14570] [Q&A] [answered]
### Bug description

When using prisma.$queryRaw i got this message
`prisma.$queryRaw is not a function`

I've already used prisma.$queryRaw on another project with prisma/client@3.15.2.
Here I'm using prisma/client@4.1.0

I don't get what I'm doing wrong here

Thanks

### How to reproduce

/

### Prisma information

/

### Environment & setup

- OS: Mac OS
- Database: MySQL
- Node.js version: v16.15.0

### Prisma Version

```
prisma                  : 4.1.1
@prisma/client          : 4.1.1
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules/@prisma/engines/migration-engine-darwin-arm64)
Introspection Engine    : introspection-core 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules/@prisma/engines/introspection-engine-darwin-arm64)
Format Binary           : prisma-fmt 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules/@prisma/engines/prisma-fmt-darwin-arm64)
Default Engines Hash    : 8d8414deb360336e4698a65aa45a1fbaf1ce13d8
Studio                  : 0.469.0
Preview Features        : fullTextSearch, fullTextIndex
```

**Comment by janpio** [comment, marked as answer]
$queryRaw definitely still exists: https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw

Do other Prisma Client queries work?

**Reply by luminous8** [reply]
yes, I'm using the usual prisma.table.findMany without any issue
I want to use $queryRaw for a complex query.

When I console.log({ prisma }) I can see all the table so everything is good on this side

**Reply by janpio** [reply]
Can you maybe, just for reproduction, create a new project that is kind of similar to your current one, and then try to use the $queryRaw API there? Does it work? If not, can you put that project on GitHub?

**Reply by luminous8** [reply]
It still doesn't work

I've put it on Github > https://github.com/luminous8/debug-prisma

**Reply by luminous8** [reply]
Did you had the chance to have a look? 
Thanks

**Reply by austincrim** [reply]
In `db.js`, try exporting like this:
```ts
module.exports = { prisma }
```

**Reply by luminous8** [reply]
Still got the error message

**Reply by austincrim** [reply]
Sorry, forgot to mention you will also need to import like so:
```ts
const { prisma } = require('./db.js')
```

**Reply by luminous8** [reply]
It works, thanks!
Would you mind explaining why/how it solved the issue?

**Reply by austincrim** [reply]
`module.exports = { prisma }` exports an **object** with a property of `prisma` that represents the Prisma Client.

Importing like `const { prisma } = require('./db.js')` destructures that property into a variable `prisma`.

Importing like `const prisma = require('./db.js')` means `prisma` is the full object, meaning you would have to use it like `await prisma.prisma.user.findFirst()` because the actual `PrismaClient` is on the `prisma` property.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
